src/App.tsx

```diff
@@ -8,7 +8,51 @@
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
 export const App: React.FC = () => {
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+  const [selectOption, setSelectOption] = useState('all');
+  const [searchInput, setSearchInput] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+      })
+      .catch(error => {
+        // eslint-disable-next-line no-console
+        console.error('Error fetching todos:', error);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    const byTitle = todo.title
+      .toLowerCase()
+      .includes(searchInput.toLowerCase());
+
+    const statusSelect = (() => {
+      switch (selectOption) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    })();
+
+    return byTitle && statusSelect;
```

Comment:
The filtering logic is correct, but ensure that the `selectOption` values ('active', 'completed', etc.) are consistent with the options provided in the `TodoFilter` component to avoid any mismatches.